icu4c/source/i18n/decNumber.h

```diff
@@ -86,7 +86,7 @@
                          /* range: -1999999997 through 999999999      */
     uint8_t bits;        /* Indicator bits (see above)                */
                          /* Coefficient, from least significant unit  */
-    decNumberUnit lsu[DECNUMUNITS];
+    decNumberUnit lsu[DECNUMUNITS+2];
```

Comment:
@sffc if this is necessary, then it should be upstreamed

Reply:
ICU4C *is* the upstream for decNumber.

However, I would like to understand more about this change.

1. Could you provide a test case please that exercises the case where DECNUMUNITS is not sufficient?
2. Instead of adding `+2` here, I think it would be cleaner if you instead changed the definition of `DECNUMUNITS` itself, a few lines earlier, and updated the comment to explain why you need the extra 2 units.